scripts/exact-example.jl

```diff
@@ -0,0 +1,237 @@
+        [i == cld(M, 2) || i == cld(M, 2) + 1 ? 1 : 0 for i in 1:M],
+        [i == cld(M, 2) ? 1 : 0 for i in 1:M],
+    )
+end
+nothing # hide
+
+# The Hamiltonians in Rimu are dimensionless. We use the following function to convert the
+# units to physical units (``\frac{\hbar^2}{mL}``). This function also handles
+# renormalization, which can greatly improve convergence with increasing ``M``, as we'll see
+# below.
+
+function convert_units(g, M; renormalize=false)
+    if renormalize
+        g = g / (1 + g / (π^2 * M))
+    end
+    t = float(M^2/2)
+    u = float(t*2/M*g)
+    return (; u, t)
+end
+nothing # hide
+
+# We know the energy of this system with ``g = -10``, so we save that in a constant.
+
+const reference_energy = -15.151863462651115
+nothing # hide
+
+# We will be comparing a lattice model, a lattice-renormalized model and the transcorrelated
+# model. Note that the `dispersion=continuum_dispersion` argument passed to
+# `HubbardMom1D` is there to get a quadratic dispersion - without it, the dispersion of the
+# Hamiltonian would be a cosine, which would introduce an offset in the energy. As a first
+# step, we can create the three Hamiltonians with 5 modes.
+
+M = 5
+u, t = convert_units(-10, M)
+u_ren, _ = convert_units(-10, M, renormalize=true)
+address = init_address(M)
+lattice = HubbardMom1D(address; u, t, dispersion=continuum_dispersion)
+trcorr = Transcorrelated1D(address; v=u, t)
+renorm = HubbardMom1D(address; u=u_ren, t, dispersion=continuum_dispersion)
+nothing # hide
+
+# ## The BasisSetRep
+
+# As we'll see later, there are two ways to construct the matrices from Hamiltonians
+# directly, but they both use `BasisSetRep` under the hood. The `BasisSetRep`, when called
+# with a Hamiltonian and optionally a starting address, constructs the sparse matrix of the
+# system and its basis. The starting address defaults to the one that was used to initialize
+# the Hamiltonian. `BasisSetRep` only returns the part of the matrix that is accessible from
+# this starting address through non-zero offdiagonal elements.
+
+lattice_bsr = BasisSetRep(lattice)
+
+# To access the matrix or basis, access the `sm` and `basis` fields, respectively.
+
+lattice_bsr.sm
+
+#
+
+lattice_bsr.basis
+
+# When the basis is not needed, we can use `Matrix` or `sparse` directly.
+
+Matrix(trcorr)
+
+#
+
+sparse(renorm)
+
+# ## Computing eigenvalues
+
+# Now that we have a way of constructing matrices from Hamiltonians, we can use standard
+# Julia functionality to diagonalize them. To make the computation more interesting, we
+# will use a different method to diagonalize each Hamiltonian at different values of `M` and
+# compare the results.
+
+# For the first, we will use `LinearAlgebra`'s `eigvals` (also see `eigs` and `eigvecs`)
+# function. This is only recommended for very small or dense Hamiltonians. It tends
+# to be the fastest method, but storing the full matrix requires the most memory.
+
+lattice_energies = map(5:2:20) do M
```

Comment:
```suggestion
m_range = 5:2:20 # range of values for the number of single-particle modes 

lattice_energies = map(m_range) do M
```
The idea would be to re-use `m_range` for the other examples and the plotting as well.